snakemake/script.py

```diff
@@ -109,25 +109,22 @@ def log_fmt_shell(self, stdout=True, stderr=True, append=False):
         return lookup[(stdout, stderr, append)].format(self.log)
 
 
-class REncoder:
-    """Encoding Pyton data structures into R."""
-
-    @classmethod
-    def encode_numeric(cls, value):
-        if value is None:
-            return "as.numeric(NA)"
-        return str(value)
+class BaseEncoder:
+    """shared class for encoding a value. An Encoder subclass should have
+       defined: language, none_value, true_value, and false_value, all strings,
+       along with methods encode_dict, and encode_list.
+    """
 
     @classmethod
     def encode_value(cls, value):
         if value is None:
-            return "NULL"
+            return cls.none_value
```

Comment:
This also means that each encoder class provides it's own true, false, and none values.